File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int dp[] = new int[nums.length];
        dp[0] = 1;
        int result=0;
        for(int i=1;i<nums.length;i++){
        dp[i] = 1;
            for(int j=0;j<i;j++){
                if(nums[i] > nums[j]){
                dp[i] = Math.max(dp[i],dp[j]+1);
                // System.out.println("nums[j]"+nums[j]);
                }
            }
        }
        for(int i =0;i<dp.length;i++){
            result = Math.max(result,dp[i]);
        }
    return result;
    }
}

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)
        res = []
        for i in nums:
            low = 0
            high = len(res)
            while low < high:
                mid = low + (high - low) // 2
                if res[mid] >= i:
                    high = mid
                else:
                    low = mid + 1
            if len(res) == low:
                res.append(i)
            else:
                res[low] = i
        return len(res) 

        